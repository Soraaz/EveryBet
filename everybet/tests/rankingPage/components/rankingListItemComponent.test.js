import React from "react";
import renderer from "react-test-renderer";

import RankingListItemComponent from "../../../src/rankingPage/components/rankingListItemComponent";
import User from "../../../src/classes/user";

import rankingListItemStyle from "../../../src/rankingPage/styleSheets/rankingListItemStyleSheet";

describe("RankingListItemComponent", () => {
	describe("rendering", () => {
		let user = {
			key: "1",
			id: 42,
			login: "login",
			coins: 69,
			avatar: null
		};

		it("Should render correctly the first place", () => {
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly the second place", () => {
			user.key = "2";
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly the third place", () => {
			user.key = "3";
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly any place starting the fourth", () => {
			user.key = "4";
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly the first place with orange border for corresponding id", () => {
			User.id = 42;
			user.key = "1";
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly the second place with orange border for corresponding id", () => {
			user.key = "2";
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly the third place with orange border for corresponding id", () => {
			user.key = "3";
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly any place starting the fourth with orange border for corresponding id", () => {
			user.key = "4";
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly the first place with a custom avatar", () => {
			User.id = undefined;
			user.key = "1";
			user.avatar = "avatar";
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly the second place with a custom avatar", () => {
			user.key = "2";
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly the third place with a custom avatar", () => {
			user.key = "3";
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly any place starting the fourth with a custom avatar", () => {
			user.key = "4";
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly the first place with orange border for corresponding id and a custom avatar", () => {
			User.id = 42;
			user.key = "1";
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly the second place with orange border for corresponding id and a custom avatar", () => {
			user.key = "2";
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly the third place with orange border for corresponding id and a custom avatar", () => {
			user.key = "3";
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly any place starting the fourth with orange border for corresponding id and a custom avatar", () => {
			user.key = "4";
			const tree = renderer
				.create(<RankingListItemComponent user={user}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});
	});

	describe("functionalities", () => {
		it("Should return the gold style for the first place", () => {
			expect(RankingListItemComponent._rankColor(1)).to.be.equal(rankingListItemStyle.gold);
		});

		it("Should return the silver style for the second place", () => {
			expect(RankingListItemComponent._rankColor(2)).to.be.equal(rankingListItemStyle.silver);
		});

		it("Should return the bronze style for the third place", () => {
			expect(RankingListItemComponent._rankColor(3)).to.be.equal(rankingListItemStyle.bronze);
		});

		it("Should return the white style for any place starting the fourth", () => {
			expect(RankingListItemComponent._rankColor(4)).to.be.equal(rankingListItemStyle.white);
		});
	});
});