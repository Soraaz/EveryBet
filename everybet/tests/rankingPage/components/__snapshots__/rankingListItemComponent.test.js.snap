// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RankingListItemComponent rendering Should render correctly any place starting the fourth 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#FFFFFF7F",
      },
      "",
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    4
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/user_placeholder.png",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;

exports[`RankingListItemComponent rendering Should render correctly any place starting the fourth with a custom avatar 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#FFFFFF7F",
      },
      "",
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    4
  </Text>
  <Image
    source={
      Object {
        "method": "GET",
        "uri": "avatar",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;

exports[`RankingListItemComponent rendering Should render correctly any place starting the fourth with orange border for corresponding id 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#FFFFFF7F",
      },
      Object {
        "borderColor": "#F8BC4C",
        "borderStyle": "solid",
        "borderWidth": 5,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    4
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/user_placeholder.png",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;

exports[`RankingListItemComponent rendering Should render correctly any place starting the fourth with orange border for corresponding id and a custom avatar 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#FFFFFF7F",
      },
      Object {
        "borderColor": "#F8BC4C",
        "borderStyle": "solid",
        "borderWidth": 5,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    4
  </Text>
  <Image
    source={
      Object {
        "method": "GET",
        "uri": "avatar",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;

exports[`RankingListItemComponent rendering Should render correctly the first place 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#D4AF377F",
      },
      "",
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    1
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/user_placeholder.png",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;

exports[`RankingListItemComponent rendering Should render correctly the first place with a custom avatar 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#D4AF377F",
      },
      "",
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    1
  </Text>
  <Image
    source={
      Object {
        "method": "GET",
        "uri": "avatar",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;

exports[`RankingListItemComponent rendering Should render correctly the first place with orange border for corresponding id 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#D4AF377F",
      },
      Object {
        "borderColor": "#F8BC4C",
        "borderStyle": "solid",
        "borderWidth": 5,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    1
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/user_placeholder.png",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;

exports[`RankingListItemComponent rendering Should render correctly the first place with orange border for corresponding id and a custom avatar 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#D4AF377F",
      },
      Object {
        "borderColor": "#F8BC4C",
        "borderStyle": "solid",
        "borderWidth": 5,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    1
  </Text>
  <Image
    source={
      Object {
        "method": "GET",
        "uri": "avatar",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;

exports[`RankingListItemComponent rendering Should render correctly the second place 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#C4CACE7F",
      },
      "",
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    2
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/user_placeholder.png",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;

exports[`RankingListItemComponent rendering Should render correctly the second place with a custom avatar 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#C4CACE7F",
      },
      "",
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    2
  </Text>
  <Image
    source={
      Object {
        "method": "GET",
        "uri": "avatar",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;

exports[`RankingListItemComponent rendering Should render correctly the second place with orange border for corresponding id 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#C4CACE7F",
      },
      Object {
        "borderColor": "#F8BC4C",
        "borderStyle": "solid",
        "borderWidth": 5,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    2
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/user_placeholder.png",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;

exports[`RankingListItemComponent rendering Should render correctly the second place with orange border for corresponding id and a custom avatar 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#C4CACE7F",
      },
      Object {
        "borderColor": "#F8BC4C",
        "borderStyle": "solid",
        "borderWidth": 5,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    2
  </Text>
  <Image
    source={
      Object {
        "method": "GET",
        "uri": "avatar",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;

exports[`RankingListItemComponent rendering Should render correctly the third place 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#CD7F327F",
      },
      "",
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    3
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/user_placeholder.png",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;

exports[`RankingListItemComponent rendering Should render correctly the third place with a custom avatar 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#CD7F327F",
      },
      "",
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    3
  </Text>
  <Image
    source={
      Object {
        "method": "GET",
        "uri": "avatar",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;

exports[`RankingListItemComponent rendering Should render correctly the third place with orange border for corresponding id 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#CD7F327F",
      },
      Object {
        "borderColor": "#F8BC4C",
        "borderStyle": "solid",
        "borderWidth": 5,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    3
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/user_placeholder.png",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;

exports[`RankingListItemComponent rendering Should render correctly the third place with orange border for corresponding id and a custom avatar 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 35,
        "marginTop": 15,
      },
      Object {
        "backgroundColor": "#CD7F327F",
      },
      Object {
        "borderColor": "#F8BC4C",
        "borderStyle": "solid",
        "borderWidth": 5,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    3
  </Text>
  <Image
    source={
      Object {
        "method": "GET",
        "uri": "avatar",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 10,
        "marginRight": "auto",
      }
    }
  >
    login
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "System",
        "fontSize": 15,
        "marginLeft": "auto",
      }
    }
  >
    69
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/betcoin_icon.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginLeft": 10,
        "marginRight": 30,
        "width": 30,
      }
    }
  />
</View>
`;
